src/common/mfu_daos.c

```diff
@@ -1313,3 +1326,2840 @@ int mfu_flist_copy_daos(
+            path_len = snprintf(number_str, ATTR_NAME_LEN, "%d",
+                                (int)(*ak_index));
+            if (path_len >= ATTR_NAME_LEN) {
+                MFU_LOG(MFU_LOG_ERR, "number_str is too long");
+                rc = 1;
+                goto out;
+            }
+            path_len = snprintf(attr_num_str, ATTR_NAME_LEN, "-%d", attr_num);
+            if (path_len >= ATTR_NAME_LEN) {
+                MFU_LOG(MFU_LOG_ERR, "attr number str is too long");
+                rc = 1;
+                goto out;
+            }
+            path_len = snprintf(attr_name, ATTR_NAME_LEN, "%s%s%s", "A-",
+                                number_str, attr_num_str);
+            if (path_len >= ATTR_NAME_LEN) {
+                MFU_LOG(MFU_LOG_ERR, "attr name is too long");
+                rc = 1;
+                goto out;
+            }
+            hdf5->attr_dims[0] = nalloc;
+            hdf5->attr_dspace = H5Screate_simple(1, hdf5->attr_dims,
+                                                 NULL);
+            if (hdf5->attr_dspace < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to create attr");
+                rc = 1;
+                goto out;
+            }
+            hdf5->selection_attr = H5Acreate2(hdf5->rx_dset,
+                                              attr_name,
+                                              hdf5->rx_dtype,
+                                              hdf5->attr_dspace,
+                                              H5P_DEFAULT,
+                                              H5P_DEFAULT);
+            if (hdf5->selection_attr < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to create selection attr");
+                rc = 1;
+                goto out;
+            }   
+            status = H5Awrite(hdf5->selection_attr, hdf5->rx_dtype,
+                              encode_buf);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to write attr");
+                rc = 1;
+                goto out;
+            }
+            status = H5Aclose(hdf5->selection_attr);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to close attr");
+                rc = 1;
+                goto out;
+            }
+            status = H5Sclose(hdf5->rx_memspace);
+            if (status < 0) {
+                MFU_LOG(MFU_LOG_ERR, "failed to close rx_memspace");
+                rc = 1;
+                goto out;
+            }
+            mfu_free(&encode_buf);
+            attr_num++;
+        }
+    }
+out:
+    if (rc != 0) {
+        status = H5Aclose(hdf5->selection_attr);
+        if (status < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to close attr");
+            rc = 1;
+            goto out;
+        }
+        status = H5Sclose(hdf5->rx_memspace);
+        if (status < 0) {
+            MFU_LOG(MFU_LOG_ERR, "failed to close rx_memspace");
+            rc = 1;
+            goto out;
+        }
```

Comment:
@dsikich These goto statements cause an infinite loop. Maybe we don't need to log or check the status here at all, since this is for an error condition anyway.

```suggestion
        H5Aclose(hdf5->selection_attr);
        H5Sclose(hdf5->rx_memspace);
```

Reply:
@dsikich Actually, it also seems that HDF5 itself throws errors if trying to close something that isn't valid, so I think we might also need to only try to close if it was opened/valid/etc., but I'm not sure the best way.

Reply:
@daltonbohning I think we can maybe check if these are > 0, and if they are then they should be closed